src/main/java/net/minestom/server/tab/TabListManager.java

```diff
@@ -0,0 +1,116 @@
+package net.minestom.server.tab;
+
+import net.minestom.server.entity.Player;
+import net.minestom.server.network.packet.server.play.PlayerInfoPacket;
+import net.minestom.server.utils.PacketUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+public class TabListManager {
+    private final Set<TabList> tabLists = new CopyOnWriteArraySet<>();
+    private TabList defaultTabList = this.createTabList();
+
+    /**
+     * Creates a new TabList
+     *
+     * @return a newly created TabList
+     */
+    public @NotNull TabList createTabList() {
+        TabList tablist = new TabList();
+        this.tabLists.add(tablist);
+        return tablist;
+    }
+
+    /**
+     * Gets the registered TabLists
+     *
+     * @return the registered {@link TabList}s
+     */
+    public @NotNull Set<TabList> getTabLists() {
+        return this.tabLists;
+    }
+
+    /**
+     * This TabList is assigned to a player when they log in.
+     *
+     * @return the default TabList
+     */
+    public @NotNull TabList getDefaultTabList() {
```

Comment:
As far as I can tell there isn't a way to provide a default tab list per player. May be worth either exposing it in `PlayerSpawnEvent` (or an earlier one), or changing this to be a `TabListProvider` or something where the default ignores the player.